### PR TITLE
Adv training

### PR DESCRIPTION
**Pull request by rithik83**
Solves issue #23. I have described as well as I can the issue, let me know if it's confusing.

Currently seems as though FGSM training actually works. Will test PGD training soon.

Edit: PGD training seems to work too. Only thing is if the adversary in train-time is too strong, the model doesn't seem to want to learn anything.

Also based on a few runs, what i notice looks like it makes sense. Classical training performs poorly on both FGSM and PGD adversaries. FGSM training does decent against FGSM adversary but poorly against PGD. PGD training did well against both. Checks out

### CODE REVIEW

File: src/training/adversarial_training.jl [hunk 1]
```diff
@@ -11,12 +11,10 @@ function vanilla_train(
     batch_size;
     loss = logitcrossentropy,
     opt = Adam,
-    min_label = 0,
-    max_label = 9,
 )
     θ = Flux.params(model)
     vanilla_losses = []
-    train_loader = DataLoader((x_train, y_train), batchsize = batch_size, shuffle = true)
+    train_loader = DataLoader((x_train, y_train), batchsize = batch_size, shuffle = true) |> gpu
```
Comment (pat-alt) [inline src/training/adversarial_training.jl:17]: Should this be `Flux.get_device()` to keep it general? https://fluxml.ai/Flux.jl/stable/guide/gpu/#Using-device-objects

File: src/training/adversarial_training.jl [hunk 2]
```diff
@@ -55,59 +53,60 @@ function adversarial_train(
...
     iterations = 10,
     attack_method = :FGSM,
-    min_label = 0,
-    max_label = 9,
     clamp_range = (0, 1),
 )
     adv_losses = []
     θ = Flux.params(model)
-    train_loader = DataLoader((x_train, y_train), batchsize = batch_size, shuffle = true)
+    train_loader = DataLoader((x_train, y_train), batchsize = batch_size, shuffle = true) |> gpu
```
Comment (pat-alt) [inline src/training/adversarial_training.jl:60]: Same as above

**Review by pat-alt — COMMENTED** [review]
LGTM although you could perhaps still add tests for the training functions? Guess local and manual testing is also sufficient for now though

File: src/training/adversarial_training.jl (hunk 1 above)
Comment (rithik83) [reply src/training/adversarial_training.jl:17]: Neat, didn't know that was a thing. I'll change this, and I'll also add some tests for the train, and the attacks (simple ones for now to check if the perturbation is within bounds)

**Review by pat-alt — APPROVED** [review]

**Pull request merged**